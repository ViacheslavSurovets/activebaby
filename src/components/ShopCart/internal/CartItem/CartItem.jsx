import React from 'react';
import {
  CartItemContainer,
  CartItemDetails,
  CartItemImage,
  CartItemPrice,
  CartItemText
} from './styles';
import PropTypes from 'prop-types';

const CartItem = ( { item } ) => {
    const { images, price, title, quantity } = item;
  return (
    <CartItemContainer>
      <CartItemImage src={ images[ 0 ] } alt={ title } />
      <CartItemDetails>
        <CartItemText>{ title }</CartItemText>
        <CartItemPrice>
          { quantity } x { price }BYN
        </CartItemPrice>
      </CartItemDetails>
    </CartItemContainer>
  );
};

export default CartItem;

CartItem.propTypes = {
  item: PropTypes.object
};
