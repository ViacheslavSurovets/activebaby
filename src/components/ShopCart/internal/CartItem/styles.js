import styled from 'styled-components';

export const CartItemContainer = styled.div`
  display: flex;
  height: 8rem;
  width: 95%;
  margin: .5rem auto;
`;

export const CartItemImage = styled.img`
  width: 30%;
`;

export const CartItemDetails = styled.div`
  display:flex;
  width: 70%;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding: 10px 20px;
`;

export const CartItemText = styled.span`

`;

export const CartItemPrice = styled.span`

`;
