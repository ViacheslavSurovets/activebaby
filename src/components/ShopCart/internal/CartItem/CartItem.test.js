import React from 'react';
import { shallow } from 'enzyme';
import CartItem from './CartItem';


describe ( 'CartItem', () => {

  const mockItem = {
    images: [],
    price: 'string',
    title: 'string',
    quantity: 'string'
  };

  it ( 'SearchMobile Snapshot', () => {
    expect ( shallow ( <CartItem item={ mockItem } /> ) ).toMatchSnapshot ();
  } );
} );
