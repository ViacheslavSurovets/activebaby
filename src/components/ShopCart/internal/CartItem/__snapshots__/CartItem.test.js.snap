// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem SearchMobile Snapshot 1`] = `ShallowWrapper {}`;
